bazel/foreign_cc/nghttp2.patch

```diff
@@ -15,15 +27,31 @@ index e70d2d49..f9c5340e 100644
  endif()
  # AC_TYPE_UINT8_T
  # AC_TYPE_UINT16_T
-diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
-index 17e422b2..a10f82a5 100644
---- a/lib/CMakeLists.txt
-+++ b/lib/CMakeLists.txt
-@@ -56,6 +56,7 @@ if(HAVE_CUNIT OR ENABLE_STATIC_LIB)
-     COMPILE_FLAGS "${WARNCFLAGS}"
-     VERSION ${LT_VERSION} SOVERSION ${LT_SOVERSION}
-     ARCHIVE_OUTPUT_NAME nghttp2
-+    ARCHIVE_OUTPUT_NAME nghttp2_static
+@@ -385,6 +389,15 @@ else()
    )
-   target_compile_definitions(nghttp2_static PUBLIC "-DNGHTTP2_STATICLIB")
-   if(ENABLE_STATIC_LIB)
+ endif()
+ 
++if(ENABLE_STATIC_CRT)
```

Comment:
Can you open a PR in upstream too for this change?

Reply:
I've opened three representing the all of the changes in nghttp2.patch (including one correction).

Correct ssize_t on Windows
https://github.com/nghttp2/nghttp2/pull/1395

Avoid filename collision of static and dynamic lib
https://github.com/nghttp2/nghttp2/pull/1394
(Dropping the erroneous ARCHIVE_OUTPUT_NAME nghttp2 value)

https://github.com/nghttp2/nghttp2/pull/1393
Add new flag ENABLE_STATIC_CRT for Windows